priam/src/main/java/com/netflix/priam/aws/RemoteBackupPath.java

```diff
@@ -90,27 +91,37 @@ private Path getV2Location() {
             prefix = Paths.get(prefix.toString(), keyspace, columnFamily);
         }
 
-        return Paths.get(prefix.toString(), fileName + "." + getCompression().toString());
+        return Paths.get(
+                prefix.toString(),
+                getCompression().toString(),
+                getEncryption().toString(),
+                fileName);
     }
 
     private void parseV2Location(String remoteFile) {
         Path remoteFilePath = Paths.get(remoteFile);
         parseV2Prefix(remoteFilePath);
-        assert remoteFilePath.getNameCount() >= 6;
-        type = BackupFileType.valueOf(remoteFilePath.getName(3).toString());
-        setLastModified(Instant.ofEpochMilli(Long.parseLong(remoteFilePath.getName(4).toString())));
+        assert remoteFilePath.getNameCount() >= 8;
```

Comment:
Our prod and test fleet Priams do not run with assertions enabled, so having this assertion would not help, either enable assertions ( which I am not big fan of enabling it in prod) or validations would help